app/controllers/events/view/index.js

```diff
@@ -1,51 +1,76 @@
 
-export default Controller.extend({
-  sponsorsColumns: [
-    {
-      propertyName   : 'logo-url',
-      template       : 'components/ui-table/cell/cell-image',
-      title          : 'Logo',
-      disableSorting : true
-    },
-    {
-      propertyName : 'name',
-      title        : 'Name'
-    },
-    {
-      propertyName   : 'type',
-      template       : 'components/ui-table/cell/cell-sponsor-sanitize',
-      title          : 'Type',
-      disableSorting : true
-    },
-    {
-      propertyName : 'level',
-      template     : 'components/ui-table/cell/cell-sponsor-sanitize',
-      title        : 'Level'
-    },
-    {
-      title          : 'Options',
-      template       : 'components/ui-table/cell/cell-sponsor-options',
-      disableSorting : true
-    }
-  ],
-  actions: {
-    deleteSponsor(sponsor) {
-      this.set('isLoading', true);
-      sponsor.destroyRecord()
-        .then(() => {
-          this.notify.success(this.l10n.t('Sponsor has been deleted successfully.'));
-          this.get('model.sponsors').removeObject(sponsor);
-        })
-        .catch(() => {
-          this.notify.error(this.l10n.t('An unexpected error has occurred.'));
-        })
-        .finally(() => {
-          this.set('isLoading', false);
-        });
-    },
-    editSponsor() {
-      this.transitionToRoute('events.view.edit.sponsors');
-    }
+export default class extends Controller.extend(EmberTableControllerMixin) {
+  per_page = 100;
+
+  sort_dir = 'ASC';
+
+  @computed()
+  get sponsorsColumns() {
+    return [
+      {
+        name          : 'Logo',
+        valuePath     : 'logo-url',
+        cellComponent : 'ui-table/cell/cell-image'
+      },
+      {
+        name            : 'Name',
+        valuePath       : 'name',
+        headerComponent : 'tables/headers/sort',
+        isSortable      : true
+
+      },
+      {
+        name            : 'Type',
+        valuePath       : 'type',
+        cellComponent   : 'ui-table/cell/cell-sponsor-sanitize',
+        extraValuePaths : ['type'],
+        options         : {
+          PropName: 'type'
+        }
+      },
+      {
+        name            : 'Level',
+        valuePath       : 'level',
+        cellComponent   : 'ui-table/cell/cell-sponsor-sanitize',
+        extraValuePaths : ['level'],
+        options         : {
+          PropName: 'level'
+        }
+      },
+      {
+        name          : 'Options',
+        valuePath     : 'options',
+        cellComponent : 'ui-table/cell/cell-sponsor-options',
+        actions       : {
+          editSponsor   : this.editSponsor.bind(this),
+          deleteSponsor : this.deleteSponsor.bind(this)
+        }
+      }
+    ];
   }
-});
+
+     @action
+  deleteSponsor(sponsor) {
+    this.set('isLoading', true);
+    sponsor.destroyRecord()
+      .then(() => {
+        this.notify.success(this.l10n.t('Sponsor has been deleted successfully.'));
+        this.get('model.sponsors').removeObject(sponsor);
```

Comment:
use the latest method introduced in mixin

Reply:
Used the `refreshModel`